ruby/examples/app/controllers/user_registration_controller.rb

```diff
@@ -0,0 +1,31 @@
+class UserRegistrationController < ApplicationController
+  def new
+    @user = UserRegistration.new
+  end
+
+  def create
+    request = RegisterUser::Request.new(*user_params)
+    response = RegisterUser.new.register(request)
```

Comment:
Controllers should have the construction of use cases hidden from them. This is so they don't become dependency magnets.

```ruby
request = RegisterUser::Request.new(*user_params)
response = AcmeApplication.execute(RegisterUser, request)
```

In this case, AcmeApplication dispatches the symbol RegisterUser to some code somewhere that knows where to get the dependencies from, but ensures your controller is unaware of the dependencies of your use case